lib/surro-gate/proxy.rb

```diff
@@ -111,16 +111,21 @@ def thread_stop
     end
 
     def reactor
-      # Atomically get an array of readable/writable monitors
-      monitors = @mutex.synchronize { @selector.select(0.1) || [] }
+      # Atomically get an array of readable monitors while also polling for writables
+      monitors = @mutex.synchronize do
+        @writer.select(0.1)
+        @reader.select(0.1) || []
+      end
       # Call each transmission proc and collect the results
-      callers = monitors.map { |m| m.value.call }
-      # Sleep for a short time if there was no transmission
-      sleep(0.1) if callers.none? && monitors.any?
+      monitors.map { |m| m.value.call }
+      # Sleep for a short time if there are no sockets
+      sleep(0.1) if @reader.empty? && @writer.empty?
```

Comment:
Is 0.1 used for a reason?  I'm not sure what value would be best but 0.1 is pretty long delay unless it's not going to happen that often.